### PR TITLE
Update 1.7.2

### PR DESCRIPTION
**Pull request by blmaier**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by blmaier** [comment]
Hi @conda-forge/core team, this package appears unmaintained. Could I get a review on this update?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -36,22 +35,26 @@ outputs:
...
       build:
         - {{ compiler('c') }}
-        - make
+        - {{ stdlib('c') }}
+        - bison
+        - flex
+        - pkg-config
       host:
         - {{ pin_subpackage("libfdt", exact=True) }}
+        - python {{ python }}
```
Comment (h-vetinari) [inline recipe/meta.yaml]: Why is python needed here? The binaries are compiled from C, which is confirmed by the link check:
```
   INFO (dtc,bin/fdtoverlay): Needed DSO lib/libfdt.so.1 found in local/linux-64::libfdt==1.7.2=h9ecbd09_0
   INFO (dtc,bin/fdtoverlay): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_18
   INFO (dtc,bin/fdtput): Needed DSO lib/libfdt.so.1 found in local/linux-64::libfdt==1.7.2=h9ecbd09_0
   INFO (dtc,bin/fdtput): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_18
   INFO (dtc,bin/fdtdump): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_18
   INFO (dtc,bin/convert-dtsv0): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_18
   INFO (dtc,bin/fdtget): Needed DSO lib/libfdt.so.1 found in local/linux-64::libfdt==1.7.2=h9ecbd09_0
   INFO (dtc,bin/fdtget): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_18
   INFO (dtc,bin/dtc): Needed DSO lib/libyaml-0.so.2 found in conda-forge/linux-64::yaml==0.2.5=h7f98852_2
   INFO (dtc,bin/dtc): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_18
```
I think this should just be removed
```suggestion
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -64,62 +67,44 @@ outputs:
...
-    script: install.sh
+    build:
+      script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
     requirements:
       build:
         - {{ compiler('c') }}
-        - make
-        - swig
+        - {{ stdlib('c') }}
+        - swig <4.2
```
Comment (h-vetinari) [inline recipe/meta.yaml]: Can you comment why the cap on swig is necessary?

File: recipe/meta.yaml [hunk 3]
```diff
@@ -131,26 +116,39 @@ outputs:
...
         - test -f $PREFIX/include/libfdt.h
         - test -f $PREFIX/include/fdt.h
         - test -f $PREFIX/include/libfdt_env.h
-        - test -f $PREFIX/lib/libfdt-{{ version }}${SHLIB_EXT}
         - test -f $PREFIX/lib/libfdt.a
         - test -f $PREFIX/lib/libfdt${SHLIB_EXT}
-        - test -f $PREFIX/lib/libfdt${SHLIB_EXT}.{{ major_ver }}
+        - test -f $PREFIX/lib/libfdt.{{ major_ver }}${SHLIB_EXT}  # [osx]
+        - test -f $PREFIX/lib/libfdt${SHLIB_EXT}.{{ major_ver }}  # [linux]
+        - test -f $PREFIX/lib/libfdt${SHLIB_EXT}.{{ version }}  # [linux]
```
Comment (h-vetinari) [inline recipe/meta.yaml:140]: I'm assuming the switch to the more standard SOVER-handling on osx is due to the switch to meson. Given that `libfdt` advertises a minor-level pin in the run-export, and we're going from 1.6 to 1.7, this should be fine. For completeness, I check who uses `libfdt`, and it's actually only used by this feedstock itself.
```
$mamba repoquery whoneeds -c conda-forge -p linux-64 libfdt

Executing the query libfdt

conda-forge/noarch                                  17.4MB @   1.5MB/s 11.4s
conda-forge/linux-64                                40.0MB @   2.6MB/s 15.7s


 Name Version Build        Depends     Channel  Subdir
-------------------------------------------------------
 dtc  1.6.1   h166bdaf_0   conda-forge linux-64
 dtc  1.6.1   h166bdaf_1   conda-forge linux-64
 dtc  1.6.1   h166bdaf_2   conda-forge linux-64
```

**Review by h-vetinari — COMMENTED** [review]
This looks very good, thanks a lot for stepping up @blmaier! Indeed, it looks like @timsnyder hasn't had any activity in conda-forge since ~March. The only thing I think we need to fix is that the binaries in `dtc` don't need (and shouldn't have) a python dependence.

File: recipe/meta.yaml (hunk 3 above)
Comment (blmaier) [reply recipe/meta.yaml:140]: Thank you for verifying that. How did you check the run-export?

File: recipe/meta.yaml (hunk 1 above)
Comment (blmaier) [reply recipe/meta.yaml]: I agree both the dtc and libfdt outputs shouldn't need python. I had tried removing this before, and when doing so the templating can't find PYTHON. Do you know why that would be?

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1089416&view=logs&j=d818e662-7181-5ce9-b265-50d03742b7af&t=367317f3-9b77-59ac-9e12-e56370c98acd&l=607

File: recipe/meta.yaml (hunk 3 above)
Comment (h-vetinari) [reply recipe/meta.yaml:140]: See here
https://github.com/conda-forge/dtc-package-feedstock/blob/dff90d43a55dea85754ecca9472c7aa3094c58bf/recipe/meta.yaml#L129-L130

**Comment by h-vetinari** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml (hunk 2 above)
Comment (blmaier) [reply recipe/meta.yaml]: I was hitting some bug with swig that I fixed by switching to 4.2, but it builds with swig 4.3 now. Removed the limit.

**Comment by h-vetinari** [comment]
One thing that's a bit silly is that each job builds the same `libfdt` and `dtc`
```
# job for 3.9
anaconda upload \
    /home/conda/feedstock_root/build_artifacts/linux-64/libfdt-1.7.2-hb9d3cd8_0.conda \
    /home/conda/feedstock_root/build_artifacts/linux-64/dtc-1.7.2-hb9d3cd8_0.conda \
    /home/conda/feedstock_root/build_artifacts/linux-64/pylibfdt-1.7.2-py39h8cd3c5a_0.conda
# 3.10
anaconda upload \
    /home/conda/feedstock_root/build_artifacts/linux-64/libfdt-1.7.2-hb9d3cd8_0.conda \
    /home/conda/feedstock_root/build_artifacts/linux-64/dtc-1.7.2-hb9d3cd8_0.conda \
    /home/conda/feedstock_root/build_artifacts/linux-64/pylibfdt-1.7.2-py310ha75aee5_0.conda
# 3.11
anaconda upload \
    /home/conda/feedstock_root/build_artifacts/linux-64/libfdt-1.7.2-hb9d3cd8_0.conda \
    /home/conda/feedstock_root/build_artifacts/linux-64/dtc-1.7.2-hb9d3cd8_0.conda \
    /home/conda/feedstock_root/build_artifacts/linux-64/pylibfdt-1.7.2-py311h9ecbd09_0.conda
# 3.12
anaconda upload \
    /home/conda/feedstock_root/build_artifacts/linux-64/libfdt-1.7.2-hb9d3cd8_0.conda \
    /home/conda/feedstock_root/build_artifacts/linux-64/dtc-1.7.2-hb9d3cd8_0.conda \
    /home/conda/feedstock_root/build_artifacts/linux-64/pylibfdt-1.7.2-py312h66e93f0_0.conda
```
This is not the end of the world, in the sense that all of them have exactly colliding build strings, meaning it'll only get uploaded once (by whichever job finishes first).

What would be better would be to collapse them all into a single job per platform that builds the lib and then 4 x `pylibfdt`. Given that he build scripts are already separated here, that should actually be pretty easy.

**Comment by h-vetinari** [comment]
@conda-forge-admin, please rerender

**Comment by h-vetinari** [comment]
> One thing that's a bit silly is that each job builds the same `libfdt` and `dtc`

Contrast to all-in-one after updates 🙃 
```
anaconda upload \
    /home/conda/feedstock_root/build_artifacts/linux-64/libfdt-1.7.2-hb9d3cd8_0.conda \
    /home/conda/feedstock_root/build_artifacts/linux-64/dtc-1.7.2-hb9d3cd8_0.conda \
    /home/conda/feedstock_root/build_artifacts/linux-64/pylibfdt-1.7.2-py312h66e93f0_0.conda \
    /home/conda/feedstock_root/build_artifacts/linux-64/pylibfdt-1.7.2-py310ha75aee5_0.conda \
    /home/conda/feedstock_root/build_artifacts/linux-64/pylibfdt-1.7.2-py39h8cd3c5a_0.conda \
    /home/conda/feedstock_root/build_artifacts/linux-64/pylibfdt-1.7.2-py311h9ecbd09_0.conda
```

**Review by h-vetinari — APPROVED** [review]
Thank you for stepping up! 💪🙏

**Comment by blmaier** [comment]
Thanks for consolidating the build outputs, that's a nice improvement 😄. Thank you for reviewing my PR

**Pull request merged**